imagepipeline-backends/imagepipeline-okhttp3/src/main/java/com/facebook/imagepipeline/backends/okhttp3/OkHttpNetworkFetcher.java

```diff
@@ -54,16 +54,23 @@ public OkHttpNetworkFetchState(
   private static final String TOTAL_TIME = "total_time";
   private static final String IMAGE_SIZE = "image_size";
 
-  private final OkHttpClient mOkHttpClient;
+  private final Call.Factory mCallFactory;
 
   private Executor mCancellationExecutor;
 
   /**
    * @param okHttpClient client to use
    */
   public OkHttpNetworkFetcher(OkHttpClient okHttpClient) {
-    mOkHttpClient = okHttpClient;
-    mCancellationExecutor = okHttpClient.dispatcher().executorService();
+    this(okHttpClient, okHttpClient.dispatcher().executorService());
+  }
+
+  /**
+   * @param callFactory client to use
+   */
```

Comment:
You should also add a JavaDoc the second parameter here and update this one to be a bit more descriptive